<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>AutoPanorama</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="mainWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabMain">
       <attribute name="title">
        <string>Files</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QFileWidget" name="filesListWidget">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="iconSize">
           <size>
            <width>200</width>
            <height>130</height>
           </size>
          </property>
          <property name="movement">
           <enum>QListView::Snap</enum>
          </property>
          <property name="resizeMode">
           <enum>QListView::Adjust</enum>
          </property>
          <property name="layoutMode">
           <enum>QListView::Batched</enum>
          </property>
          <property name="gridSize">
           <size>
            <width>210</width>
            <height>150</height>
           </size>
          </property>
          <property name="viewMode">
           <enum>QListView::IconMode</enum>
          </property>
          <property name="batchSize">
           <number>1</number>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="buttonshorizontalLayout">
          <item>
           <widget class="QPushButton" name="addFiles_button">
            <property name="text">
             <string>Add files</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeFiles_button">
            <property name="text">
             <string>Remove selected</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="18" column="0">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QLabel" name="nbands_label">
            <property name="text">
             <string>Bands</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="sharpness_label">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Sharpness</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="18" column="1">
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QSpinBox" name="nbands_spinbox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="sharpness_spinbox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.020000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="13" column="0">
         <widget class="QLabel" name="seamfinderres_label">
          <property name="text">
           <string>Seam finder resolution</string>
          </property>
         </widget>
        </item>
        <item row="15" column="0">
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="13" column="1">
         <widget class="QDoubleSpinBox" name="seamfinderres_spinbox">
          <property name="suffix">
           <string> Mpx</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="featuresfinder_combobox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>AKAZE</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>ORB</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="25" column="1">
         <widget class="QDoubleSpinBox" name="compositingres_spinbox">
          <property name="prefix">
           <string/>
          </property>
          <property name="suffix">
           <string> Mpx (zero for full resolution)</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
        <item row="10" column="1">
         <widget class="QComboBox" name="expcomp_combobox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Blocks Gain</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Gain</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="cut_label">
          <property name="text">
           <string>Warp mode</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QComboBox" name="warpmode_combobox">
          <item>
           <property name="text">
            <string>Spherical</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Cylindrical</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Plane</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="5">
         <widget class="QComboBox" name="wavecorkind_combobox">
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Horizontal</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Vertical</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="4">
         <widget class="QLabel" name="wavecorkind_label">
          <property name="text">
           <string>Wave correction</string>
          </property>
         </widget>
        </item>
        <item row="10" column="0">
         <widget class="QLabel" name="expcom_label">
          <property name="text">
           <string>Exposure compensator</string>
          </property>
         </widget>
        </item>
        <item row="11" column="0">
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="7" column="1">
         <widget class="QComboBox" name="bundleadj_combobox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Ray</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Reproj</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="featuresfinder_label">
          <property name="text">
           <string>Features finder</string>
          </property>
         </widget>
        </item>
        <item row="17" column="0">
         <widget class="QLabel" name="blendertype_label">
          <property name="text">
           <string>Blender type</string>
          </property>
         </widget>
        </item>
        <item row="17" column="1">
         <widget class="QComboBox" name="blendertype_combobox">
          <item>
           <property name="text">
            <string>Multiband</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Feather</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="regres_label">
          <property name="text">
           <string>Registration resolution</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="regres_spinbox">
          <property name="suffix">
           <string> Mpx</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>0.600000000000000</double>
          </property>
         </widget>
        </item>
        <item row="7" column="4">
         <widget class="QLabel" name="confth_label">
          <property name="text">
           <string>Confidence threshold</string>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QLabel" name="exposure_label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Exposure</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="featuresmatcher_label">
          <property name="text">
           <string>Features matcher</string>
          </property>
         </widget>
        </item>
        <item row="16" column="0">
         <widget class="QLabel" name="blending_label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Blending</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="featuresmatcher_combobox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Best of 2 nearest</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="23" column="0">
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="24" column="0">
         <widget class="QLabel" name="compositing_label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Compositing</string>
          </property>
         </widget>
        </item>
        <item row="7" column="5">
         <widget class="QDoubleSpinBox" name="confth_spinbox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="5" column="4">
         <widget class="QLabel" name="featuresmatcherconf_label">
          <property name="text">
           <string>Match confidence</string>
          </property>
         </widget>
        </item>
        <item row="14" column="0">
         <widget class="QLabel" name="seamfindermode_label">
          <property name="text">
           <string>Seam finder type</string>
          </property>
         </widget>
        </item>
        <item row="25" column="0">
         <widget class="QLabel" name="compositingres_label">
          <property name="text">
           <string>Compositing resolution</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="registration_label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Registration</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="bundleadj_label">
          <property name="text">
           <string>Bundle Adjuster type</string>
          </property>
         </widget>
        </item>
        <item row="5" column="5">
         <widget class="QDoubleSpinBox" name="featuresmatcherconf_spinbox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.650000000000000</double>
          </property>
         </widget>
        </item>
        <item row="12" column="0">
         <widget class="QLabel" name="seaming_label">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Seaming</string>
          </property>
         </widget>
        </item>
        <item row="14" column="1">
         <widget class="QComboBox" name="seamfindermode_combobox">
          <item>
           <property name="text">
            <string>Graph cut color</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Graph cut gradient</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Voronoi</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="10" column="4">
         <widget class="QLabel" name="blocksize_label">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Block size</string>
          </property>
         </widget>
        </item>
        <item row="10" column="5">
         <widget class="QSpinBox" name="blocksize_spinbox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>8</number>
          </property>
          <property name="maximum">
           <number>128</number>
          </property>
          <property name="value">
           <number>32</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabProgress">
       <attribute name="title">
        <string>Progress</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="sizeAdjustPolicy">
           <enum>QAbstractScrollArea::AdjustToContents</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>898</width>
             <height>524</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="autoFillBackground">
            <bool>true</bool>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <layout class="QVBoxLayout" name="tabProgressLayout"/>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="2" column="0">
     <layout class="QHBoxLayout" name="layoutOutput">
      <item>
       <widget class="QLineEdit" name="output_filename_lineedit">
        <property name="text">
         <string>panorama</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="extension_combobox">
        <property name="editable">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>.jpg</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>.png</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonMakePanorama">
        <property name="text">
         <string>Make panorama</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="output_label">
      <property name="text">
       <string>Output file name</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>940</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QFileWidget</class>
   <extends>QListWidget</extends>
   <header>src/qfilewidget.h</header>
   <slots>
    <slot>selectAndAddFiles()</slot>
    <slot>removeSelected()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonMakePanorama</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onMakePanoramaClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>589</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blendertype_combobox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onBlenderTypeChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>expcomp_combobox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onExposureCompensatorChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addFiles_button</sender>
   <signal>clicked()</signal>
   <receiver>filesListWidget</receiver>
   <slot>selectAndAddFiles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>586</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeFiles_button</sender>
   <signal>clicked()</signal>
   <receiver>filesListWidget</receiver>
   <slot>removeSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>590</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSelectDirClicked()</slot>
  <slot>onMakePanoramaClicked()</slot>
  <slot>onBlenderTypeChange()</slot>
  <slot>onExposureCompensatorChange()</slot>
 </slots>
</ui>
